File: halcyon-web/src/main/resources/static/js/service/hr/hrEmployeeService.js
/**
 * author:Marshal
 * description:service层
 */
app.service("hrEmployeeService", function ($http) {
    //base crud
    this.query = function (pageNum, pageSize, condition) {
        return $http.post("../hr/employee/query?pageNum=" + pageNum + "&pageSize=" + pageSize, condition);
    }

    this.save = function (hrEmployee) {
        return $http.post("../hr/employee/submit", hrEmployee);
    }

    this.delete = function (selectedIds) {
        return $http.get("../hr/employee/remove?selectedIds=" + selectedIds);
    }

    this.queryById = function (id) {
        return $http.get("../hr/employee/selectByEmployeeId?employeeId=" + id);
    }

    this.getEmpOptions = function () {
        return $http.get("../hr/employee/getEmpOptions");
    }
});

File: halcyon-base/src/main/java/com/marshal/halcyon/base/hr/service/HrEmployeeService.java
package com.marshal.halcyon.base.hr.service;


import com.marshal.halcyon.core.service.BaseService;
import com.marshal.halcyon.base.hr.entity.HrEmployee;

import java.util.List;
import java.util.Map;

public interface HrEmployeeService extends BaseService<HrEmployee> {

    HrEmployee selectByUserId(Long userId);

    List<Map> getEmpOptions();

    List<Map> getEmpCodeOptions();

}
